Source/WarpX.cpp

```
@@ -2112,6 +2112,14 @@ WarpX::BackwardCompatibility ()
             ablastr::warn_manager::WarnPriority::low);
     }
 
+    if (pp_particles.contains("photon_species")){
+        ablastr::warn_manager::WMRecordWarning("Species",
+            "particles.photon_species is deprecated and may be removed in the future. "
+            "It is recommended to initialize photon particles by setting their "
+            "'species_type' to 'photon', instead.",
+            ablastr::warn_manager::WarnPriority::low);
+    }
```

I had a question here, but I didn't ask quickly enough. The warning says that the option "may be removed in the future". Does this mean that it hasn't been removed yet? If it has been removed, i.e., either we don't parse it or we don't use it even if we parse it, shouldn't the warning say that the option "has been removed" instead?

---

My understanding is that it hasn't been removed yet. (i.e. the previous input script using `particles.photon_species` will still work, but raise the warning.)

---

Okay, I don't know what policy we follow, but I would have just removed it at once and raised an error instead of a warning. The plan now would be that one of us remembers to actually remove it at some point in the future? Or do we just leave all old input options hidden but still functional usually? I don't remember our policy here.

---

I can confirm what @RemiLehe said: I was reluctant to introduce a breaking change, so with this PR only a low-priority warning is generated and the instruction is still parsed and used as before (with a minor difference in the case of explicitly setting the mass of a species to zero, which is now forbidden). I am open to discuss other solutions and I acknowledge that we don't have a clear protocol to deal with deprecated features. I think that, if we decide to introduce this possibility, we should remove deprecated features after ~ 3-6 months.